game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/BattleStepsTest.java

```diff
@@ -0,0 +1,2199 @@
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.NAVAL_BOMBARDMENT;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.REMOVE_CASUALTIES;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.REMOVE_PREFIX;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.REMOVE_SNEAK_ATTACK_CASUALTIES;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.REMOVE_UNESCORTED_TRANSPORTS;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.SELECT_CASUALTIES;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.SELECT_FIRST_STRIKE_CASUALTIES;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.SELECT_NAVAL_BOMBARDMENT_CASUALTIES;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.SELECT_PREFIX;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.SUBMERGE_SUBS_VS_AIR_ONLY;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.SUBS_SUBMERGE;
+import static games.strategy.triplea.delegate.battle.BattleStepStrings.SUBS_WITHDRAW;
+import static games.strategy.triplea.delegate.battle.steps.BattleSteps.FIRE;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.engine.data.properties.GameProperties;
+import games.strategy.triplea.Constants;
+import games.strategy.triplea.attachments.TechAttachment;
+import games.strategy.triplea.attachments.UnitAttachment;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class BattleStepsTest {
+
+  @Mock GameData mockGameData;
+  @Mock GameProperties mockGameProperties;
+  @Mock Function<Collection<Unit>, Collection<Unit>> mockGetDependentUnits;
+  @Mock Supplier<Collection<Territory>> getAttackerRetreatTerritories;
+
+  @Mock
+  BiFunction<GamePlayer, Collection<Unit>, Collection<Territory>> getEmptyOrFriendlySeaNeighbors;
+
+  @Mock Territory mockBattleSite;
```

Comment:
nit: I used to personally have a style of prefixing mocks with `mock`, I think that style is a bit outdated now and the preference is to omit the `mock` prefix on mock variable names. IE: this variable would be just `battleSite`.